### DISCUSSION TITLE
Howto support different local schema versions

### DISCUSSION THREAD

**Discussion by vvmruder** [redhat-developer/vscode-yaml#753] [Q&A]
Maybe I really miss some point here. But I have the following situation:

I locally store different versions of the same schema for a YAML in the way:

- `.../schema/master/service.schema.json`
- `.../schema/1.0.5/service.schema.json`
- ...
- `.../schema/0.0.4/service.schema.json`
- ...

I then have several projects using this schemas. And it works well. Big THANK YOU for that.

But over time it happens that project-a's `service.yml` is of version 0.0.4 and project-b's `service.yml` is of version 1.0.5. To be able to clarify that, I have an attribute `schema_version` in my `service.yml`.

Problem now: Whenever I want vscode to validate against my schema in correct version I need to edit my configuration in the `settings.json` because there I define:

```
{
    "yaml.schemas": {
        "file:///.../schema/master/service.schema.json": [
            "service.yml"
        ]
    }
}
```

So my questions are:
- Am I doing wrong here?
- Is there a better way?
- Maybe even to directly use information of version number from my `service.yml` via place holder in settings?

**Comment by JPinkney** [comment]
The only way to do it currently is to make a vscode extension that uses the [register contributor api](https://github.com/redhat-developer/vscode-yaml/wiki/Extension-API#register-contributor) and sends back schemas based off of what schema_version is in that file

**Comment by vvmruder** [comment]
Thank you for this hint. I will try that. Looks promising.

**Comment by JPinkney** [comment]
If you go that route and have any questions let me know! I'm hoping to create a sample project that shows people how to do this in the coming months
